cloud/scope/machine.go

```diff
@@ -168,15 +168,25 @@ func (m *MachineScope) CreateMachine() (*vpcv1.Instance, error) {
 
 // DeleteMachine deletes the vpc machine associated with machine instance id.
 func (m *MachineScope) DeleteMachine() error {
-	options := &vpcv1.DeleteInstanceOptions{}
-	options.SetID(m.IBMVPCMachine.Status.InstanceID)
-	_, err := m.IBMVPCClient.DeleteInstance(options)
+	listOptions := &vpcv1.ListInstancesOptions{}
+	instances, _, err := m.IBMVPCClient.ListInstances(listOptions)
 	if err != nil {
-		record.Warnf(m.IBMVPCMachine, "FailedDeleteInstance", "Failed instance deletion - %v", err)
-	} else {
-		record.Eventf(m.IBMVPCMachine, "SuccessfulDeleteInstance", "Deleted Instance %q", m.IBMVPCMachine.Name)
+		return err
 	}
-	return err
+	for _, instance := range instances.Instances {
+		if *instance.ID == m.IBMVPCMachine.Status.InstanceID {
```

Comment:
in our case this `m.IBMVPCMachine.Status.InstanceID` is empty, not sure we should even do a listing and get into this loop if its empty. Other providers are listing my tags if instance id empty considering it is not yet set by the controller and I'm not if we want introduce that mechanism here or not!

Our source of truth for instance creation is `Status.InstanceID`, hence if this is not created then safely we can return early saying instance is not created yet.

Reply:
Agreed with your point but what I have observed is safely returning the delete call saying ID not yet set so can't trigger delete seems to hang forever too, as the machines already enters the delete phase:( (let me verify it again quickly)

Reply:
Fixed.